packages/connection/src/Manager.php

```diff
@@ -43,6 +160,313 @@ public function initialize( $methods ) {
+				return new \WP_Error(
+					'unknown_user',
+					sprintf( 'User %d does not exist', $user_id ),
+					compact( 'signature_details' )
+				);
+			}
+		}
+
+		$token = $this->get_access_token( $user_id, $token_key, false );
+		if ( is_wp_error( $token ) ) {
+			$token->add_data( compact( 'signature_details' ) );
+			return $token;
+		} elseif ( ! $token ) {
+			return new \WP_Error(
+				'unknown_token',
+				sprintf( 'Token %s:%s:%d does not exist', $token_key, $version, $user_id ),
+				compact( 'signature_details' )
+			);
+		}
+
+		$jetpack_signature = new \Jetpack_Signature( $token->secret, (int) \Jetpack_Options::get_option( 'time_diff' ) );
+		// phpcs:disable WordPress.Security.NonceVerification.Missing
+		if ( isset( $_POST['_jetpack_is_multipart'] ) ) {
+			$post_data   = $_POST;
+			$file_hashes = array();
+			foreach ( $post_data as $post_data_key => $post_data_value ) {
+				if ( 0 !== strpos( $post_data_key, '_jetpack_file_hmac_' ) ) {
+					continue;
+				}
+				$post_data_key                 = substr( $post_data_key, strlen( '_jetpack_file_hmac_' ) );
+				$file_hashes[ $post_data_key ] = $post_data_value;
+			}
+
+			foreach ( $file_hashes as $post_data_key => $post_data_value ) {
+				unset( $post_data[ "_jetpack_file_hmac_{$post_data_key}" ] );
+				$post_data[ $post_data_key ] = $post_data_value;
+			}
+
+			ksort( $post_data );
+
+			$body = http_build_query( stripslashes_deep( $post_data ) );
+		} elseif ( is_null( $this->raw_post_data ) ) {
+			$body = file_get_contents( 'php://input' );
+		} else {
+			$body = null;
+		}
+		// phpcs:enable
+
+		$signature = $jetpack_signature->sign_current_request(
+			array( 'body' => is_null( $body ) ? $this->raw_post_data : $body )
+		);
+
+		$signature_details['url'] = $jetpack_signature->current_request_url;
+
+		if ( ! $signature ) {
+			return new \WP_Error(
+				'could_not_sign',
+				'Unknown signature error',
+				compact( 'signature_details' )
+			);
+		} elseif ( is_wp_error( $signature ) ) {
+			return $signature;
+		}
+
+		$timestamp = (int) $_GET['timestamp'];
+		$nonce     = stripslashes( (string) $_GET['nonce'] );
+
+		// Use up the nonce regardless of whether the signature matches.
+		if ( ! $this->add_nonce( $timestamp, $nonce ) ) {
+			return new \WP_Error(
+				'invalid_nonce',
+				'Could not add nonce',
+				compact( 'signature_details' )
+			);
+		}
+
+		// Be careful about what you do with this debugging data.
+		// If a malicious requester has access to the expected signature,
+		// bad things might be possible.
+		$signature_details['expected'] = $signature;
+
+		if ( ! hash_equals( $signature, $_GET['signature'] ) ) {
+			return new \WP_Error(
+				'signature_mismatch',
+				'Signature mismatch',
+				compact( 'signature_details' )
+			);
+		}
+
+		/**
+		 * Action for additional token checking.
+		 *
+		 * @since 7.5.0
+		 *
+		 * @param Array $post_data request data.
+		 * @param Array $token_data token data.
+		 */
+		return apply_filter(
```

Comment:
Typo.

```suggestion
		return apply_filters(
```

Reply:
Thank you, I have found that one because of breaking tests too.